appserver/web/web-core/src/main/java/org/apache/catalina/core/StandardWrapperValve.java

```diff
@@ -135,7 +140,23 @@ public int invoke(Request request, Response response)
         RequestFacade hreq = (RequestFacade) request.getRequest(true);
         HttpServletResponse hres =
             (HttpServletResponse) response.getResponse();
+        
+        String contextPath = context.getServletContext().getContextPath();
+        
+        if (!contextPath.isEmpty()) {
+            ServiceLocator services = org.glassfish.internal.api.Globals.getDefaultHabitat();
+            ApplicationRegistry appRegistry = services.<ApplicationRegistry>getService(ApplicationRegistry.class);
+            applicationInfo = appRegistry.get(context.getServletContext().getContextPath().replace("/", ""));
```

Comment:
This is only true, when context path matches application (i. e. not overriden in glassfish-web, not part of an EAR, ...)